tools/udevadm

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+# NOTE: Since this runs on OpenWRT, it needs to restrict itself to commands
+#       which are compatible with ash which is the sh used there.
+#
+# This script is intended to be used on the OpenWRT platform which doesn't
+# have udev installed. It basically produces output compatible with running
+# the command "udevadm info -e" which is what the node serialport module
+# does here: https://github.com/node-serialport/node-serialport/blob/master/lib/bindings/linux-list.js
+#
+# This inspiration for this came from the Python pyserial file:
+# https://github.com/pyserial/pyserial/blob/master/serial/tools/list_ports_linux.py
+#
+# This use of this allows serial port enumeration, needed for several of the
+# gateway adapters, to work under OpenWRT.
+
+enumerate_device() {
+  NAME=$(basename $1)
+  DEVICE_PATH=$(readlink -f /sys/class/tty/${NAME}/device)
+  if [ ! -e "${DEVICE_PATH}" ]; then
+    return
+  fi
+  echo "P: $(echo ${DEVICE_PATH} | cut -c5-)/tty/${NAME}"
+  echo "N: ${NAME}"
+  echo "E: DEVNAME=$1"
+  SUBSYSTEM=$(basename $(readlink -f ${DEVICE_PATH}/subsystem))
+  if [ "${SUBSYSTEM}" == "usb-serial" ]; then
```

Comment:
I ran this on the Pi. It didn't work unless I changed `==` to `=` (ditto on line 28). `==` isn't supported by all shells.

Reply:
Interesting. It seems dash (which is the default sh under ubuntu) doesn't support it, but ash (the shell used under openwrt) does.

I happen to have my sh set to bash on my dev machine, which is why I didn't notice it.

I'll change it anyways.